### PR TITLE
chore(platform):Refactor GraphExecution naming clash and remove unused Graph Execution functions

### PR DESCRIPTION
**Pull request by majdyz**
This is a follow-up of https://github.com/Significant-Gravitas/AutoGPT/pull/8752

There are several APIs and functions related to graph execution that are unused now.
There is also confusion about the name of `GraphExecution` that exists in graph.py & execution.py.

### Changes 🏗️

* Renamed `GraphExecution` in `execution.py` to `GraphExecutionEntry`, this is only used as a queue entry for execution.
* Removed unused `get_graph_execution` & `list_executions` in `execution.py`.
* Removed `with_run` option on `get_graph` function in `graph.py`.
* Removed `GraphMetaWithRuns`
* Removed exposed functions only for testing.
* Removed `executions` fields in Graph model.

#### For code changes:
- [ ] I have clearly listed my changes in the PR description
- [ ] I have made a test plan
- [ ] I have tested my changes according to the test plan:
  
  - [ ] ...

<details>
  <summary>Example test plan</summary>
  
  - [ ] Create from scratch and execute an agent with at least 3 blocks
  - [ ] Import an agent from file upload, and confirm it executes correctly
  - [ ] Upload agent to marketplace
  - [ ] Import an agent from marketplace and confirm it executes correctly
  - [ ] Edit an agent from monitor, and confirm it executes correctly
</details>

#### For configuration changes:
- [ ] `.env.example` is updated or already compatible with my changes
- [ ] `docker-compose.yml` is updated or already compatible with my changes
- [ ] I have included a list of my configuration changes in the PR description (under **Changes**)

<details>
  <summary>Examples of configuration changes</summary>

  - Changing ports
  - Adding new services that need to communicate with each other
  - Secrets or environment variable changes
  - New or infrastructure changes such as databases
</details>

### CODE REVIEW

File: autogpt_platform/frontend/src/app/page.tsx [hunk 1]
```diff
@@ -87,9 +87,7 @@ const Monitor = () => {
         selectedFlow={selectedFlow}
         onSelectFlow={(f) => {
           setSelectedRun(null);
-          setSelectedFlow(
-            f.id == selectedFlow?.id ? null : (f as GraphMetaWithRuns),
-          );
+          setSelectedFlow(f.id == selectedFlow?.id ? null : (f as GraphMeta));
```
Comment (kcze) [inline autogpt_platform/frontend/src/app/page.tsx:90]: Maybe outside the scope but it'd be good to rename all `*flow` to `*graph` and `*run` to `*execution` for consistency.

File: autogpt_platform/frontend/src/lib/autogpt-server-api/types.ts [hunk 2]
```diff
@@ -200,7 +200,7 @@ export type GraphExecution = {
   graph_version: number;
 };
 
-/* backend/data/graph.py:Graph = GraphMeta & GraphMetaWithRuns & Graph */
+/* backend/data/graph.py:Graph = GraphMeta & Graph */
```
Comment (kcze) [inline autogpt_platform/frontend/src/lib/autogpt-server-api/types.ts]:
```suggestion
```

File: autogpt_platform/frontend/src/lib/autogpt-server-api/types.ts [hunk 3]
```diff
@@ -212,10 +212,6 @@ export type GraphMeta = {
   output_schema: BlockIOObjectSubSchema;
 };
 
-export type GraphMetaWithRuns = GraphMeta & {
-  executions: GraphExecution[];
-};
-
 export type Graph = GraphMeta & {
```
Comment (kcze) [inline autogpt_platform/frontend/src/lib/autogpt-server-api/types.ts:215]: Now this is a mirror
```suggestion
/* Mirror of backend/data/graph.py:Graph */
export type Graph = GraphMeta & {
```

**Review by kcze — CHANGES_REQUESTED** [review]
Good cleanup 😊

File: autogpt_platform/frontend/src/app/page.tsx (hunk 1 above)
Comment (majdyz) [reply autogpt_platform/frontend/src/app/page.tsx:90]: Yeah I'll keep this PR less invasive here.

**Review by kcze — APPROVED** [review]

**Pull request merged**